cadence/contracts/bridge/FlowEVMBridge.cdc

```diff
@@ -376,23 +407,42 @@ access(all) contract FlowEVMBridge {
     ///
     /// @returns The EVMAddress of the deployed contract
     ///
-    access(self) fun deployERC721(_ forNFTType: Type): EVM.EVMAddress {
-        let name: String = FlowEVMBridgeUtils.getContractName(fromType: forNFTType)
+    access(self)
+    fun deployERC721(_ forNFTType: Type): EVM.EVMAddress {
+        // Retrieve the Cadence type's defining contract name, address, & its identifier
+        var name = FlowEVMBridgeUtils.getContractName(fromType: forNFTType)
             ?? panic("Could not contract name from type: ".concat(forNFTType.identifier))
-        let identifier: String = forNFTType.identifier
-        let cadenceAddressStr: String = FlowEVMBridgeUtils.getContractAddress(fromType: forNFTType)?.toString()
+        let identifier = forNFTType.identifier
+        let cadenceAddress = FlowEVMBridgeUtils.getContractAddress(fromType: forNFTType)
             ?? panic("Could not derive contract address for token type: ".concat(identifier))
+        // Assign a default symbol
+        var symbol = "BRDG"
+        // Borrow the ViewResolver to attempt to resolve the EVMBridgedMetadata view
+        let viewResolver = getAccount(cadenceAddress).contracts.borrow<&{ViewResolver}>(name: name)!
+        var contractURI = ""
+        if let bridgedMetadata = viewResolver.resolveContractView(
+                resourceType: forNFTType,
+                viewType: Type<CrossVMNFT.EVMBridgedMetadata>()
+            ) as! CrossVMNFT.EVMBridgedMetadata? {
+            name = bridgedMetadata.name
+            symbol = bridgedMetadata.symbol
+            contractURI = bridgedMetadata.uri.uri()
```

Comment:
There's no guarantees that a token would implement `baseURI`, especially since the value can be overwritten. To cover all cases, I can take for granted that `tokenURI` should always resolve to something.
WRT contract vs token level metadata, I think the `EVMBridgedMetadata` view is generic enough to cover both, resolving either depending on where the view is being resolved. For example:
```cadence
let tokenView = nft.resolveView(
    Type<CrossVMNFT.EVMBridgedMetadata>()
  )! as! CrossVMNFT.EVMBridgedMetadata
// Here, I would expect token-level metadata
tokenView.uri.uri()

let contractView = ExampleNFT.resolveView(
    Type<CrossVMNFT.EVMBridgedMetadata>()
  )! as! CrossVMNFT.EVMBridgedMetadata
// Here I would expect contract-level metadata
contractView.uri.uri()
```

I guess we could distinguish these two by differing types, but they contents of the two would largely be the same.